app/assets/javascripts/discourse/app/lib/topic-list-tracker.js

```diff
@@ -1,8 +1,25 @@
 import { Promise } from "rsvp";
 let model, currentTopicId;
 
+let highestReadCache = {};
+
 export function setTopicList(incomingModel) {
   model = incomingModel;
+
+  model?.topics?.forEach((topic) => {
+    let highestReadFromCache = highestReadCache[topic.id];
+    if (
+      highestReadCache &&
```

Comment:
That should be this, right?

```suggestion
      highestReadFromCache &&
```

btw. I'd rename this to just `highestRead` for simplicity